packages/SwingSet/src/kernel/vat-loader/transcript.js

```diff
@@ -59,13 +136,44 @@ export function makeTranscriptManager(
   let replayError;
 
   function simulateSyscall(newSyscall) {
-    const s = playbackSyscalls.shift();
-    const newReplayError = compareSyscalls(vatID, s.d, newSyscall);
-    if (newReplayError) {
-      replayError = newReplayError;
-      throw replayError;
+    while (playbackSyscalls.length) {
+      const compareError = compareSyscalls(
+        vatID,
+        playbackSyscalls[0].d,
+        newSyscall,
+      );
+
+      if (compareError === missingSyscall) {
+        // return `undefined` to indicate that this syscall cannot be simulated
+        // and needs to be performed (virtual collection metadata get)
+        return undefined;
+      }
+
+      const s = playbackSyscalls.shift();
+
+      if (!compareError) {
+        return s.response;
+      } else if (compareError !== extraSyscall) {
+        replayError = compareError;
+        break;
+      }
+
+      // Check the next transcript entry, skipping any extra syscalls recorded
+      // in the transcript (virtual collection metadata get)
+    }
+
+    if (!replayError) {
+      // Error if the vat performs a syscall for which we don't have a
+      // corresponding entry in the transcript.
+
+      // Note that if a vat performed an "allowed" vc metadata get syscall after
+      // we reach the end of the transcript, we would error instead of
+      // falling through and performing the syscall. However liveslots does not
+      // perform vc metadata get syscalls unless it needs to provide an entry
+      // to the program, which always results in subsequent syscalls.
```

Comment:
I'm not sure that's strictly true, but the chances are low, and I can't think of a clean way to deal with it, so I'm ok with this approach. The case would be something like delivery 1 doing:

```js
  baggage.set('key', makeBigScalarMapStore(..))
```

That creates the new collection, creates its initial Representative, uses that to look up the vref and stash it in baggage, then drops the Representative.

Then delivery 2 does:
```
  const p = E(presence).foo();
  const c = baggage.get('key');
  // let c fall out of scope without actually using it
  return p;
```

Delivery 2 will send `foo`, then do the vref lookup in baggage, determine that it needs a Representative for the collection, either finds it in `slotToVal` or doesn't, if it doesn't then it resurrects one (which causes the two `vatstoreGet` syscalls). But, if it really doesn't do anything else with the collection, and it doesn't do anything else syscall-worthy (i.e `p` doesn't resolve until later), then it's not going to do any other syscalls in that delivery.

If the original execution still had the `slotToVal` entry, the recorded transcript will contain the `foo` send and nothing else. If the replay finds that it *doesn't* have the `slotToVal` entry, the replay will do `foo` followed by the two `vatstoreGet`s. When the first `vatstoreGet` arrives at `simulateSyscall`, it will see `playbackSyscalls.length === 0`, and it will bypass the whole `while` loop and go directly to the final `throw replayError`.

I don't think that's going to be a common pattern (and I guess we could scan the existing transcripts for it, and convince ourselves that pismoA doesn't ever create it). But if you wanted to fix it, you might add an extra check after the `while` loop but before the `if (!replayError)`, so if we get past the `while` loop, but the syscall is the `vatstoreGet` kind, then it returns `undefined` instead of throwing.

Reply:
If you do that, it'd be worth refactoring the "is this the right kind of syscall" check into a single function, called twice from `requireIdenticalExceptStableVCSyscalls` and once from here.

... oh, except, right, that's not particularly easy to configure/activate as just replacing the `compareSyscalls` function. Hm.

Reply:
I think that edge case is not worth fixing if we're not triggering it. Worst case, it gets triggered, some validators hit a new anachrophobia, and we provide a new patch that does the more complicated check.